ntoskrnl/inbv/inbv.c

```diff
@@ -1066,13 +1092,21 @@ DisplayBootBitmap(IN BOOLEAN TextMode)
              * Save the main image palette and replace it with black palette,
              * so that we can do fade-in effect later.
              */
-            BitmapInfoHeader = (PBITMAPINFOHEADER)Screen;
-            Palette = (LPRGBQUAD)((PUCHAR)Screen + BitmapInfoHeader->biSize);
+            BitmapInfoHeader = BootLogo;
+            Palette = (LPRGBQUAD)((PUCHAR)BootLogo + BitmapInfoHeader->biSize);
             RtlCopyMemory(MainPalette, Palette, sizeof(MainPalette));
             RtlZeroMemory(Palette, sizeof(MainPalette));
 
-            /* Blit the background */
-            InbvBitBlt(Screen, 0, 0);
+            /* Draw logo */
+            InbvBitBlt(BootLogo, 225, 94);
```

Comment:
Perhaps it would make sense to calculate this position (middle of the screen) rather than hard-coding the offsets?

Reply:
Sure. Good idea.